props/saber_fett263_buttons.h

```diff
@@ -293,248 +664,3398 @@ class SaberFett263Buttons : public virtual PropBase {
+        }
+      }
+      // EVENT_THRUST
+      if (mss.y * mss.y + mss.z * mss.z < 16.0 &&
+          mss.x > 14  &&
+          fusor.swing_speed() < 150) {
+        if (millis() - thrust_begin_millis_ > 15) {
+          Event(BUTTON_NONE, EVENT_THRUST);
+          thrust_begin_millis_ = millis();
+        } 
+      } else {
+        thrust_begin_millis_ = millis();
+      }
+    }
+    DetectMenuTurn();
+    TrackPlayer();
+  }
+
+  // Fett263 Track Player
+  enum TrackMode {
+    PLAYBACK_OFF,
+    PLAYBACK_LOOP,
+    PLAYBACK_ROTATE,
+    PLAYBACK_RANDOM,
+  };
+
+  void TrackPlayer() {
+    if (track_mode_ != PLAYBACK_OFF) {
+      if (!track_player_) {
+#ifdef ENABLE_AUDIO
+        if (track_num_ == -1 && track_mode_ == PLAYBACK_LOOP) {
+          StartOrStopTrack();
+        } else {
+          if (track_num_ >= num_tracks_) track_num_ = 0;
+          switch (track_mode_) {
+            case PLAYBACK_ROTATE:
+              track_num_ += 1;
+              break;
+            case PLAYBACK_RANDOM:
+              track_num_ = rand() % num_tracks_;
+              if (track_num_ < 0) track_num_ = num_tracks_ - 1;
+              break;
+            default:
+              break;
+          }
+          char playtrack[128];
+          RunCommandAndGetSingleLine("list_current_tracks", nullptr, track_num_, playtrack, sizeof(playtrack));
+          track_player_ = GetFreeWavPlayer();
+          if (track_player_) {
+            track_player_->Play(playtrack);
+          } else {
+            STDOUT.println("No available WAV players.");
+          }
+        }
+#else
+        STDOUT.println("Audio disabled.");
+#endif
+      }
+    }
+  }
+  
+  void StopTrackPlayer() {
+    track_mode_ = PLAYBACK_OFF;
+    if (track_player_) {
+      track_player_->Stop();
+      track_player_.Free();
+    } else {
+      StartOrStopTrack();
+    }
+  }
+   
+  void DetectMenuTurn() {
+     if (menu_ || color_mode_ == CC_COLOR_LIST) {
+       if (millis() - last_rotate_millis_ > 1000) {
+        float a = fusor.angle2() - current_menu_angle_;
+        if (a > M_PI) a-=M_PI*2;
+        if (a < -M_PI) a+=M_PI*2;
+        if (a > twist_menu_ * 2/3) {
+          current_menu_angle_ += twist_menu_;
+          if (current_menu_angle_ > M_PI) current_menu_angle_ -= M_PI * 2;
+          Event(BUTTON_NONE, EVENT_TWIST_RIGHT);
+          last_rotate_millis_ = millis();
+        }
+        if (a < -twist_menu_ * 2/3) {
+          current_menu_angle_ -= twist_menu_;
+          if (current_menu_angle_ < M_PI) current_menu_angle_ += M_PI * 2;
+          Event(BUTTON_NONE, EVENT_TWIST_LEFT);
+          last_rotate_millis_ = millis();
+        }
+      }
+    }
+  }
+
+#ifdef FETT263_EDIT_MODE_MENU
+#define SUBMENUS 9
+#define COLOR_SUB 15
+#define COLOR_OPTIONS 3
+#define GESTURE_OPTIONS 12
+#define SET_SUBMENUS 6
+#define STYLE_SETTINGS 10
+#endif
+
+  enum MenuType {
+    MENU_TOP,
+    MENU_PRESET,
+    MENU_VOLUME,
+    TRACK_PLAYER,    
```

Comment:
should be MENU_TRACK_PLAYER or it's confusing...

Reply:
updated